src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy32NativeMethods.cs

```diff
@@ -0,0 +1,108 @@
+﻿/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Runtime.InteropServices;
+using MongoDB.Driver.Core.NativeLibraryLoader;
+
+namespace MongoDB.Driver.Core.Compression.Snappy
+{
+    internal static class Snappy32NativeMethods
```

Comment:
I wanted to make the change as smaller(without big changes) as possible to quickly review this PR.

Reply:
Originally I tried to refactor this logic (based on how it was implemented in my original PR), but I realized that it would be not so fast as I expected, so I will be ok to work on it later